config/routes.rb

```diff
@@ -103,15 +103,16 @@
       post 'reprocess'
     end
   end
-  resources :sample_transfers, only: [:index]
+  resources :sample_transfers, only: [:create, :index] do
+    put "confirm"
```

Comment:
I usually use `POST` for actions on a resource, and `PUT` to replace a resource (and `PATCH` for updating).